editor/sidebar/post-visibility/index.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import clickOutside from 'react-click-outside';
+import { find } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from 'i18n';
+import { Component } from 'element';
+
+/**
+ * Internal Dependencies
+ */
+import './style.scss';
+import {
+	getEditedPostAttribute,
+	getEditedPostStatus,
+	getEditedPostVisibility,
+} from '../../selectors';
+import { editPost } from '../../actions';
+
+class PostVisibility extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			opened: false,
+		};
+		this.toggleDialog = this.toggleDialog.bind( this );
+	}
+
+	toggleDialog( event ) {
+		event.preventDefault();
+		this.setState( { opened: ! this.state.opened } );
+	}
+
+	handleClickOutside() {
+		this.setState( { opened: false } );
+	}
+
+	render() {
+		const { status, visibility, password, onUpdateVisibility } = this.props;
+
+		const setPublic = () => onUpdateVisibility( visibility === 'private' ? 'draft' : status );
+		const setPrivate = () => onUpdateVisibility( 'private' );
+		const setPasswordProtected = () => onUpdateVisibility( visibility === 'private' ? 'draft' : status, password || '' );
+		const updatePassword = ( event ) => onUpdateVisibility( status, event.target.value );
+
+		const visibilityOptions = [
+			{
+				value: 'public',
+				label: __( 'Public' ),
+				info: __( 'Visible to everyone.' ),
+				changeHandler: setPublic,
+			},
+			{
+				value: 'private',
+				label: __( 'Private' ),
+				info: __( 'Only visible to site admins and editors.' ),
+				changeHandler: setPrivate,
+			},
+			{
+				value: 'password',
+				label: __( 'Password Protected' ),
+				info: __( 'Protected with a password you choose. Only those with the password can view this post.' ),
+				changeHandler: setPasswordProtected,
+			},
+		];
+		const getVisibilityLabel = () => find( visibilityOptions, { value: visibility } ).label;
+
+		// Disable Reason: The input is inside the label, we shouldn't need the htmlFor
+		/* eslint-disable jsx-a11y/label-has-for */
+		return (
+			<div className="editor-post-visibility">
+				<span>{ __( 'Visibility' ) }</span>
```

Comment:
It's is not required here, but since the parent is "flex", I like the explicitness of having flex children. It's useless though for the label.